src/deadline_worker_agent/sessions/session.py

```diff
@@ -643,6 +658,9 @@ def _start_action(self) -> None:
                 executor=self._executor,
             )
         except Exception as e:
+            if self._output_sync_target_action:
+                action_definition = self._output_sync_target_action.definition
```

Comment:
Do we also need to clear `self._output_sync_target_action` here? This exception block cancels remaining actions except for `envExit`s. So handling of those actions may be impacted.